sw/device/silicon_creator/lib/otbn_boot_services.c

```diff
@@ -124,13 +156,16 @@ rom_error_t otbn_boot_attestation_key_save(
       otbn_dmem_write(kOtbnBootModeWords, &mode, kOtbnVarBootMode));
 
   // Load the additional seed from flash info.
-  uint32_t seed_flash_offset = 0 + (additional_seed * kAttestationSeedWords);
   uint32_t seed[kAttestationSeedWords];
-  HARDENED_RETURN_IF_ERROR(
-      flash_ctrl_info_read(&kFlashCtrlInfoPageAttestationKeySeeds,
-                           seed_flash_offset, kAttestationSeedWords, seed));
+  HARDENED_RETURN_IF_ERROR(load_attestation_keygen_seed(additional_seed, seed));
+  // Pad remaining DMEM field with zeros to prevent a DMEM integrity error
+  // (since data is aligned to 256-bit words).
+  uint32_t zero_buf[16 - kAttestationSeedWords] = {0};
+  HARDENED_RETURN_IF_ERROR(otbn_dmem_write(
+      16 - kAttestationSeedWords, zero_buf,
+      kOtbnVarBootAttestationAdditionalSeed + kAttestationSeedBytes));
```

Comment:
nit: to avoid the magic number you could compute this based on the `kOtbnWideWordNumWords` constant from the OTBN driver:
```c
enum {
  /* Size of the attestation seed buffer in 32-bit words (round the attestation seed size up to the next wide word). */
  kAttestationSeedBufferWords = (kAttestationSeedWords + kOtbnWideWordNumWords - 1) / kOtbnWideWordNumWords;
};
```

```suggestion
  uint32_t zero_buf[kAttestationSeedBufferWords - kAttestationSeedWords] = {0};
  HARDENED_RETURN_IF_ERROR(otbn_dmem_write(
      ARRAYSIZE(zero_buf), zero_buf,
      kOtbnVarBootAttestationAdditionalSeed + kAttestationSeedBytes));
```